controllers/auth.go

```diff
@@ -370,6 +370,12 @@ func (c *ApiController) Login() {
 				return
 			}
 
+			if user.PasswordChangeRequired {
+				c.SetSessionUsername(user.GetId())
```

Comment:
I recommend that using another session store the tempprary user id. Because the funtion `c.SetSessionUsername()` is used for the signed in user. If user flush the web, he need not to change the password to login.

See the video:

https://github.com/casdoor/casdoor/assets/71440988/115d5c1c-0aff-4748-9f97-871b08d74327

Reply:
@leo220yuyaodog is it something like setMfaUserSession?

Reply:
> @leo220yuyaodog is it something like setMfaUserSession?

Yes

Reply:
@leo220yuyaodog thanks, please check, I've reused same way like in mfa

Reply:
@leo220yuyaodog please take a look I reused same approach like mfa,

Reply:
> @leo220yuyaodog please take a look I reused same approach like mfa, now user can't skip change password form if it requited.
> 
> Does code meet with your expectations ?
> 
> I also notice some code dublication in auth.go controller, but decide don't do any refactor in feature branch. However it could be reasonable in scope of separate technical task.

I test the code, there is a bug after change the password. After the frontend call the callback()
```go
onSuccess={() => {
                  AuthBackend.login(values, oAuthParams).then((res) => callback(res));
                }}
```
The password in oAuthParams is still the old password, so will login failed, the frontend still show successful message 
 though.

Reply:
> > @leo220yuyaodog please take a look I reused same approach like mfa, now user can't skip change password form if it requited.
> > Does code meet with your expectations ?
> > I also notice some code dublication in auth.go controller, but decide don't do any refactor in feature branch. However it could be reasonable in scope of separate technical task.
> 
> Good, the code meets my expectations.
> 
> But there is a bug after change the password. When the frontend call the login(), The password in oAuthParams is still the old password, so it will login failed, the frontend still show successful message though. Can you fix it?
> 
> ```go
> onSuccess={() => {
>                   AuthBackend.login(values, oAuthParams).then((res) => callback(res));
>                 }}
> ```
> 
>  7.24.1.mp4

Sure will fix

Reply:
@leo220yuyaodog many thanks for testing, 
I am so sorry for such a mistake, I should not have used a new password same to the old :-( 

Hope it's fixed now

Reply:
> @leo220yuyaodog Does it work for you now?

@hsluoyz  lgtm now.